platform-sdk/docs/components/componentFramework.md

```diff
@@ -0,0 +1,200 @@
+
+## Overview
+
+At the most basic level, a component in this framework is composed of a task scheduler, business logic, and wires. Each
+component has a unit of business logic that performs some action on a piece of data. The component receives the data on
+an input wire, performs the action, and optionally produces data on an output wire. The components are connected, or 
+"soldered", together by joining input and output wires of the same data type to form a data pipeline.
+
+The concurrency of the component is determined by the task scheduler type. Each component has a task scheduler which
+combines the data received on the input wire with the business logic operation to form a task. This task is then
+scheduled for execution according to the task scheduler type and implementation. Task schedulers are described in more
+detail [below](#TaskSchedulers).
+
+Here is a picture of a simple component with one input wire of type `Integer`, and one output wire of type `String`:
+
+![SimpleComponent](step4.drawio.svg)
+
+### Business Logic
+
+Business logic is defined in one or more classes that work together and accept optional data, perform an
+operation, and produce optional output. This framework allows business logic to be defined completely independently
+of the threading model and other business logic. Each unit of business logic does not know where it's input comes from
+or who consumes its output. The threading is handled by its task scheduler, and the connection of inputs and outputs is
+handled externally. Changes to concurrency and wiring have no impact on the business logic.
+
+### Wires
+
+Wires are the construct on which data flows between components. Each component can have zero or more input wires 
+(restrictions apply based on the concurrency of the task scheduler) and one or more output wires. Components that do not
+produce any data have an output void of type `Void`. Wires are strongly typed to the data they transport. Wires of the
+same type can be "soldered" or connected to determine the flow of data between components.
+
+### Task Schedulers
+
+Task schedulers are responsible for several things, but the most important job is taking data received on the input
+wire, combining it with the business logic operation to be performed on that data to create a task, and scheduling that
+task to be executed by a thread. Other responsibilities include applying back pressure (if configured) and keeping
+metrics on tasks.
+
+Data coming into the task scheduler goes through a few phases:
+
+1. Unscheduled - the data has been received by the task scheduler and a task has been created, but it has not yet been
+   scheduled for execution.
+2. Scheduled or Unprocessed - the task has been scheduled for execution and may have started execution, but has not
+   yet completed execution.
+3. Processed - the task has been executed (the method bound to the task has been called and the executing thread has
+   returned).
```

Comment:
What about error handling? Any DLQs, or the pipeline just stalls forever until a unit of data is processed?

In general, does the framework support at-most-once, or at-least-once, or exactly-once input data processing? Is this configurable per a component (or a wire) or fixed for the entire framework?

This would affect the overall performance of the framework. This may also affect how the business logic is written. So I suggest to expand on this topic in this document.

Reply:
> What about error handling? 

An uncaught exception handler can be provided, and all exceptions that occur during data processing are passed to it. If none is provided, then a default exception handler will log any exception that happens during data processing.

When there is an exception, the task scheduler does not stop or crash, it will just continue to the next item to be processed.

Reply:
> In general, does the framework support at-most-once, or at-least-once, or exactly-once input data processing? Is this configurable per a component (or a wire) or fixed for the entire framework?

Currently each task provided to a scheduler is executed exactly once.

Reply:
Thanks for the clarifications. I think it's worth adding this information to this document as it describes the contract provided by the framework.

Reply:
Excellent point, I will add this to the documentation

Reply:
Added